instrumentedProfilingLibrary/Exercises/time_profiler_constructor_destructor.cpp

```diff
@@ -0,0 +1,69 @@
+#include<iostream>
+#include<chrono>
+#include<cmath>
+
+
+using namespace std;
+
+class time_profiler{
+	public:
+		time_profiler(int N = 0): codeBlockStartTime(chrono::high_resolution_clock::now()){
```

Comment:
```suggestion
		time_profiler(std::string name): codeBlock(std::move(name)), codeBlockStartTime(chrono::high_resolution_clock::now()){
```

Reply:
Is it entirely bad practice to  use `using namespace std` instead of  `std :: string, std :: cout ...` ? From what I have seems it seems that its divided opinions.

Reply:
In .cpp files it is not bad practice. The best practice, however, is to declare what you're using in the most limited scope possible.
In this case, I think it's fine.
Putting `using namespace` clauses in header files, however, is bad because it can create unwanted conflicts and ambiguities.
For example, the `vector` class might be an array container from `std`, or a linear algebra struct from a specific `math` library you're using.